### PR TITLE
Implementation of Battle, BattleCharacter, and Move classes.

### PR DESCRIPTION
**Pull request by m-grigoriu**
This pull request is for the implementation of three Battle related entities.

Move has health damage, strength damage, caster and target attributes, each with a getter function, and one setter function for target. The caster and target instances might be unnecessary depending on how the rest of the program is implemented, but they could be useful for keeping track of character animations.

BattleCharacter has health, damage, and speed attributes, along with a list of two moves. There are getter functions for each of the three stats, and one for each of the two moves in the list. There is no implementation for the discussed 'sprite' instance yet. There are also methods to modify the health and attack stats of the entity, and the class implements Comparable for use in the Battle class.

Battle a battleStatus attribute, along with a getter function (there is no setter function, but one could be added if the option to forfeit a combat encounter is discussed). There is also a numTokens attribute, with a getter and setter function. There are ArrayList attributes for the player characters and the opponents, each with their own getter function, and a moveQueue instance which contains every character in the previous two lists sorted by their speed. There is a method to get the next character to move from the queue, which dequeues them from the list and enqueues them again. Finally, there is a method to remove a BattleCharacter from either of the two character lists and from the moveQueue. This method will update the  battleStatus if one of the two parties becomes empty.

### CODE REVIEW

**Review by sbrad044 — APPROVED** [review]

**Review by siddharthgowda — APPROVED** [review]
Looks good (only thing is that BattleCharacter needs name attribute and stat increasing functions)

**Review by siddharthgowda — CHANGES_REQUESTED** [review]
So for battleCharacter there should be a name for each character and there should be a maxHealth attribute (so upgrade tokens can increase maxHealth). Also there should be functions that can increase maxHealth, dmg, speed (so the user can increase them via upgrade tokens)

**Comment by hoodiexxx** [comment]
look pretty wholesome and functional. For getBattleStatus() method in Battle.java it might could add an if else statement and return win lose in string form.

File: src/main/java/com/mg105/entities/Battle.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        this.numTokens = 0;
+
+        this.moveQueue.addAll(this.opponents);
+        this.moveQueue.addAll(playerCharacters);
+        Collections.sort(moveQueue);
+    }
+
+    public int getBattleStatus() {
+        return battleStatus;
+    }
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/entities/Battle.java:30]: Personally, I prefer writing an if else statement and return "win", "lose" or "in progress" in string form but it is fine for returning -1,0,1 coercion.

**Review by hoodiexxx — APPROVED** [review]
wholesome coding and looks functional

**Review by siddharthgowda — APPROVED** [review]

**Review by 6167656e74323431 — APPROVED** [review]
LGTM

**Review by siddharthgowda — APPROVED** [review]
This mostly looks very good! The only issues I have is that there might be some open closed principle violations in your battleCharacter moves attributes (since you have hard coded that their is only two moves)

**Review by zhaoan12 — APPROVED** [review]
Looks good. You have many variables so I would suggest having a way to track all of them outside the program.

**Comment by 6167656e74323431** [comment]
Merged during weekly meeting.

**Pull request merged**